src/escape.rs

```diff
@@ -131,6 +155,116 @@ where
         }
         i += 1;
     }
-    w.write_all(&bytes[mark..])?;
+    w.write_all(&bytes[mark..])
+}
+
+/// Calls the given function with the index of every byte that is in
+/// [", &, <, >] and for no other byte.
+#[cfg(all(target_arch = "x86_64", feature="simd"))]
+fn scan_simd<F>(bytes: &[u8], mut offset: usize, mut callback: F) -> io::Result<()>
+    where F: FnMut(usize) -> io::Result<()>
+{
+    if bytes.is_empty() {
+        // we need this special case for when we are passed an empty subslice
+        // that is exactly on a page boundary. we would try to read on a page
+        // that is potentially not ours and crash
+        return Ok(());
+    }
+
+    let lower_vec = unsafe { _mm_set_epi8( 0, 62, 0, 60, 0, 0, 0, 0, 0, 38, 0, 0, 0, 34, 0, 127 ) };
+    let upperbound = bytes.len().saturating_sub(16);
+
+    while offset < upperbound {
+        let mut mask = unsafe {
+            let raw_ptr = transmute(bytes.as_ptr().offset(offset as isize));
+            let v = _mm_loadu_si128(raw_ptr);
+            let expected = _mm_shuffle_epi8(lower_vec, v);
```

Comment:
`_mm_shuffle_epi8` is only available in SSSE3 or later, so you need, at minimum, a `#[target_feature(enable = "ssse3")]` and you need to ensure that you're only executing this code on `x86_64` platforms that support SSSE3, probably via [`is_x86_feature_detected`](https://doc.rust-lang.org/std/arch/index.html#dynamic-cpu-feature-detection), which I don't see here at all.

Reply:
Yeah, it's an oversight - will fix!